From 09bfd5c716eeaacec06c1d326b52b47e393e4d3e Mon Sep 17 00:00:00 2001
From: Rob Clark <robclark@freedesktop.org>
Date: Mon, 13 Jun 2016 09:49:46 -0400
Subject: [PATCH 07/10] freedreno/msm: fix memory leak on ringbuffer free

Signed-off-by: Rob Clark <robclark@freedesktop.org>
---
 freedreno/msm/msm_ringbuffer.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/freedreno/msm/msm_ringbuffer.c b/freedreno/msm/msm_ringbuffer.c
index becf245..34bc2fe 100644
--- a/freedreno/msm/msm_ringbuffer.c
+++ b/freedreno/msm/msm_ringbuffer.c
@@ -353,6 +353,11 @@ static void msm_ringbuffer_destroy(struct fd_ringbuffer *ring)
 	struct msm_ringbuffer *msm_ring = to_msm_ringbuffer(ring);
 	if (msm_ring->ring_bo)
 		fd_bo_del(msm_ring->ring_bo);
+	free(msm_ring->submit.relocs);
+	free(msm_ring->submit.cmds);
+	free(msm_ring->submit.bos);
+	free(msm_ring->bos);
+	free(msm_ring->rings);
 	free(msm_ring);
 }
 
-- 
2.8.3

