---
 include/drm/drm_sarea.h |    2 	2 +	0 -	0 !
 1 file changed, 2 insertions(+)

diff -p -up ./include/drm/drm_sarea.h.1006 ./include/drm/drm_sarea.h
--- ./include/drm/drm_sarea.h.1006	2009-11-21 00:54:36.000000000 +0100
+++ ./include/drm/drm_sarea.h	2009-12-04 20:29:03.000000000 +0100
@@ -39,6 +39,8 @@
 #define SAREA_MAX                       0x2000U
 #elif defined(__ia64__)
 #define SAREA_MAX                       0x10000U	/* 64kB */
+#elif defined(__mips__)
+#define SAREA_MAX			getpagesize()
 #else
 /* Intel 830M driver needs at least 8k SAREA */
 #define SAREA_MAX                       0x2000U
